pkg/asset/agent/gencrypto/authconfig.go

```diff
@@ -128,3 +183,139 @@ func localJWTForKey(id string, privateKkeyPem string) (string, error) {
+
+	// Check if current time is past the thresholdTime time of 24 hours
+	if time.Now().After(thresholdTime) {
+		// update the secret in the cluster with a new token from asset store
+		authToken := a.AgentAuthToken
+		err = updateSecret(k8sclientset, retrievedSecret, authToken)
+		if err != nil {
+			return err
+		}
+		logrus.Debug("auth token secret regenerated and updated in the cluster")
+	} else {
+		// Update the token in asset store with the retrieved token from the cluster
+		a.AgentAuthToken = retrievedToken
+		logrus.Debugf("Auth token is valid upto %s.", expiryTime)
+	}
+	return err
+}
+
+func createSecret(k8sclientset kubernetes.Interface, newAgentAuthToken string) error {
+	// Create a Secret
+	secret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: authTokenSecretName,
+			Annotations: map[string]string{
+				"updatedAt": "", // Initially set to empty
+			},
+		},
+		Type: corev1.SecretTypeOpaque,
+		Data: map[string][]byte{
+			authTokenSecretDataKey: []byte(newAgentAuthToken),
+		},
+	}
+	_, err := k8sclientset.CoreV1().Secrets(authTokenSecretNamespace).Create(context.Background(), secret, metav1.CreateOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to create auth token secret: %w", err)
+	}
+	logrus.Debugf("Created auth token secret %s/%s", authTokenSecretNamespace, authTokenSecretName)
+
+	return nil
+}
+
+func updateSecret(k8sclientset kubernetes.Interface, retrievedSecret *corev1.Secret, newAgentAuthToken string) error {
+	retrievedSecret.Data[authTokenSecretDataKey] = []byte(newAgentAuthToken)
+	// only for informational purposes
+	retrievedSecret.Annotations["updatedAt"] = time.Now().UTC().Format(time.RFC3339)
+
+	_, err := k8sclientset.CoreV1().Secrets(authTokenSecretNamespace).Update(context.TODO(), retrievedSecret, metav1.UpdateOptions{})
+	if err != nil {
+		logrus.Fatal(err)
+	}
+	logrus.Debugf("Updated auth token secret %s/%s", authTokenSecretNamespace, authTokenSecretName)
+	return nil
+}
+
+// GetAuthTokenFromCluster returns a token string stored as the secret from the cluster.
+func GetAuthTokenFromCluster(ctx context.Context, kubeconfigPath string) (string, error) {
+	client, err := initK8sClient(kubeconfigPath)
+	if err != nil {
+		return "", err
+	}
+
+	retrievedSecret, err := client.CoreV1().Secrets(authTokenSecretNamespace).Get(ctx, authTokenSecretName, metav1.GetOptions{})
+	if err != nil {
+		return "", err
+	}
+	authToken, err := getExistingAuthToken(retrievedSecret)
+	if err != nil {
+		return "", err
+	}
+	return authToken, err
+}
+
+func getExistingAuthToken(retrievedSecret *corev1.Secret) (string, error) {
+	// Check if the secret data contains the expected key
+	existingAgentAuthToken, exists := retrievedSecret.Data[authTokenSecretDataKey]
+	if !exists {
+		logrus.Fatalf("Auth token secret %s/%s does not contain the key %s. Re-run 'add-nodes' command to create new image files(ISO/PXE files).", authTokenSecretNamespace, authTokenSecretName, authTokenSecretDataKey)
```

Comment:
Please avoid using `logrus.Fatalf`

```suggestion
		logrus.Fatalf("auth token secret %s/%s does not contain a valid token", authTokenSecretNamespace, authTokenSecretName, authTokenSecretDataKey)
```